### DISCUSSION TITLE
Running one time DML/Update scripts using Prisma

### DISCUSSION THREAD

**Discussion by dgyurics** [prisma/prisma#11833] [Q&A] [answered]
Does Prisma support running one-time DML statements, such as UPDATE, automatically?

For example, let's say we want all emails in a table to be lowercase. We make a change in our API so that all future accounts/emails are lowercase, however we want to update EXISTING emails to be lowercase too.

One way I can think of doing this is to manually create a migration folder under migrations, and generate your own migration.sql file which would contain the DML statement(s); but this seems hacky. 

FYI: Some people refer to these type of run once files as "Patch" files/scripts

**Comment by nurul3101** [comment, marked as answer]
Hey @dgyurics :wave:

Prisma doesn’t support running DML statements automatically.
Probably what you can do in your case is you could use a raw query to update the existing records.
Executing below function should lowercase all your existing emails.

```javascript
async function main() {
  const result = await prisma.$executeRaw`UPDATE "User" SET email=lower(email)`;
  console.log(result);
}
```
You can learn more about Raw Database Access from these [Guides](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#raw-queries-with-relational-databases)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
